packages/integration-tests/src/test.rs

```diff
@@ -90,24 +92,61 @@ fn proper_instantiate() -> (App, CwTemplateContract, CwTemplateContract) {
 fn post_data_request() {
     let (mut app, _wasm_bin_storage_template_contract, seda_chain_contracts_template_contract) =
         proper_instantiate();
+    let dr_binary_id: Hash = "".to_string();
+    let tally_binary_id: Hash = "".to_string();
+    let dr_inputs: Bytes = Vec::new();
+    let tally_inputs: Bytes = Vec::new();
 
-    // set arguments for post_data_request
-    let wasm_id = "wasm_id".to_string().into_bytes();
-    let wasm_args: Vec<Vec<u8>> = vec![
-        "arg1".to_string().into_bytes(),
-        "arg2".to_string().into_bytes(),
-    ];
-    let args = PostDataRequestArgs {
-        dr_id: "0xd98fb83e7f68c29c21313afd147eb6c3851d70b8d37fd75e5b78f0ecabd9f69b".to_string(), // expected
-        nonce: 1,
-        chain_id: 31337,
-        value: "test".to_string(),
-        wasm_id,
-        wasm_args,
-    };
+    let replication_factor: u16 = 3;
+
+    // set by dr creator
+    let gas_price: u128 = 10;
+    let gas_limit: u128 = 10;
+
+    // set by relayer and SEDA protocol
+    let seda_payload: Bytes = Vec::new();
+
+    let chain_id = 31337;
+    let nonce = 1;
+    let value = "test".to_string();
+    let mut hasher = Keccak256::new();
+    hash_update(&mut hasher, &chain_id);
+    hash_update(&mut hasher, &nonce);
+    hasher.update(value);
+    let binary_hash = format!("0x{}", hex::encode(hasher.finalize()));
+    let memo1: Memo = binary_hash.clone().into_bytes();
+    let payback_address: Bytes = Vec::new();
+    let mut hasher = Keccak256::new();
+    hasher.update(dr_binary_id.clone());
+    hasher.update(dr_inputs.clone());
+    hasher.update(gas_limit.to_be_bytes().clone());
```

Comment:
Well, for the bytes ones, we shouldn't need to clone or pass by reference. Just calling that method creates a new object :).